typedspark/_utils/load_table.py

```diff
@@ -35,26 +48,50 @@ def _create_schema(structtype: SparkStructType, schema_name: Optional[str] = Non
     return schema  # type: ignore
 
 
-def _extract_data_type(dtype: DataType) -> Type[DataType]:
+def _extract_data_type(dtype: DataType, name: str) -> Type[DataType]:
     """Given an instance of a ``DataType``, it extracts the corresponding
     ``DataType`` class, potentially including annotations (e.g.
     ``ArrayType[StringType]``)."""
     if isinstance(dtype, SparkArrayType):
-        element_type = _extract_data_type(dtype.elementType)
+        element_type = _extract_data_type(dtype.elementType, name)
         return ArrayType[element_type]  # type: ignore
 
     if isinstance(dtype, SparkMapType):
-        key_type = _extract_data_type(dtype.keyType)
-        value_type = _extract_data_type(dtype.valueType)
+        key_type = _extract_data_type(dtype.keyType, name)
+        value_type = _extract_data_type(dtype.valueType, name)
         return MapType[key_type, value_type]  # type: ignore
 
     if isinstance(dtype, SparkStructType):
-        subschema = _create_schema(dtype)
+        subschema = _create_schema(dtype, _to_camel_case(name))
         return StructType[subschema]  # type: ignore
 
     return type(dtype)
 
 
+def _to_camel_case(name: str) -> str:
+    """Converts a string to camel case."""
+    return "".join([word.capitalize() for word in name.split("_")])
+
+
+def create_schema(
+    dataframe: DataFrame, schema_name: Optional[str] = None
+) -> Tuple[DataSet[Schema], Type[Schema]]:
+    """This function inferres a ``Schema`` in a notebook based on a the provided ``DataFrame``.
+
+    This allows for autocompletion on column names, amongst other
+    things.
+
+    .. code-block:: python
+
+        df, Person = create_schema(df)
```

Comment:
Maybe nicer to use `ds` as the variable name for the returned value? Or something else showing that it's now a DataSet and not "just" a dataframe

Reply:
Hmm, this "problem" exists across the whole code base (in all documentation, notebooks, unit tests). I've initially chose to do this because it feels familiar. Also, it is actually bad practice to let your variable name reflect the type of the variable :p And a DataSet is still a subclass of a DataFrame, so it's not that wrong either.

We could talk about it sometime. If we really want to change it, I suggest we do it as a separate PR where we address it everywhere.